analysis/bin/vtsAnalyser.cc

```diff
@@ -213,22 +330,26 @@ void vtsAnalyser::MatchingForMC() {
 
     bool idChanged = false;
     for (unsigned int i=1;i<recoJet_.size(); ++i) {
-        if (idChanged) qjMapForMC_[recoJet_[i].idx] = -9;
-        else { 
-            if (recoJet_[0].matchedQuark != recoJet_[i].matchedQuark) idChanged = true;
-            else qjMapForMC_[recoJet_[i].idx] = -9;
-        }
+      if (idChanged) qjMapForMC_[recoJet_[i].idx] = -9;
+      else { 
+          if (recoJet_[0].matchedQuark != recoJet_[i].matchedQuark) idChanged = true;
+          else qjMapForMC_[recoJet_[i].idx] = -9;
+      }
     }
   }
 }
 
 void vtsAnalyser::HadronAnalysis() {
   std::vector<std::vector<struct HadStat>> JetCollection;
   std::vector<struct HadStat> Had;
-  struct HadStat Stat;
-
   //Reco Jet & Reco KS matching 
-  for (unsigned int j=0; j<nJet; ++j){
+  auto selectedJet = jetSelection();
+  if (selectedJet.size() == 0) {
+    cout << "Size of selectedJets is zero" << endl;
+    return;
+  }
+  for (unsigned int j=0; j<selectedJet.size();++j) {
+    j = selectedJet[j].GetFirstMother();
```

Comment:
Won't this screw up the loop? The j in the loop part won't be saved, so you'll skip to the selectedJet[j] + 1 on the next run of the loop. Use "ij" or something for the loop variable